File: src/typescript/asset-manipulations/receiving-nai-asset/receive-nai-asset.ts
import { createWaxFoundation } from '@hiveio/wax';

const waxApi = await createWaxFoundation();

// Convert the amount into `NaiAsset` for HIVE, HBD, and VESTS
const hiveAsset = waxApi.hiveSatoshis(1_000); // 1.000 HIVE
const hbdAsset = waxApi.hbdSatoshis(1_000); // 1.000 HBD
const vestsAsset = waxApi.vestsSatoshis(1_000000); // 1.000000 VESTS

console.log(`Hive Asset: ${JSON.stringify(hiveAsset)}`);
console.log(`HBD Asset: ${JSON.stringify(hbdAsset)}`);
console.log(`Vests Asset: ${JSON.stringify(vestsAsset)}`);

File: src/python/asset-manipulations/construct-nai-asset/receive_nai_asset.py
from wax import create_wax_foundation

wax = create_wax_foundation()

# Convert the amount into `NaiAsset` for HIVE, HBD, and VESTS
hive_asset = wax.hive.satoshis(amount=1_000)  # 1.000 HIVE
hbd_asset = wax.hbd.satoshis(amount=1_000)  # 1.000 HBD
vests_asset = wax.vests.satoshis(amount=1_000_000)  # 1.000000 VESTS

print(f"Hive Asset: {hive_asset}")
print(f"HBD Asset: {hbd_asset}")
print(f"Vests Asset: {vests_asset}")
